urbansearch/gathering/gathering.py

```diff
@@ -172,3 +174,64 @@ def indices_from_gz_file(self, filename):
 
             self.indices += indices
             return indices
+
+    def run_workers(self, no_of_workers, directory, queue, gz=True, opt=False):
+        """ Run workers to process indices from a directory with files
+        in parallel. All parsed indices will be added to the queue.
+
+        :no_of_workers: Number of workers that will run
+        :directory: Path to directory containing files
+        :queue: multiprocessing.Queue where the indices will be added to
+        :gz: Files are in .gz format. Default: True.
+        :opt: Determine optimal number of workers and ignore no_of_workers
+        parameter
+        """
+        if opt:
+            try:
+                no_of_workers = (cpu_count() * 2) + 1
+            except NotImplementedError:
+                logger.error("Cannot determine number of CPU's,"
+                             + "defaulting to 1 worker")
+                no_of_workers = 1
+
+        files = [_file.path for _file in os.scandir(directory)
+                 if _file.is_file()]
+
+        div_files = process_utils._divide_files(files, no_of_workers)
```

Comment:
Should the utility function actually be private if you use it in a different (this) module?